### ISSUE TITLE
Using Callbacks for Tf Estimator from Keras Model 

### ISSUE THREAD

**Issue by lysecret2** [keras-team/keras#10642]
Hey, i am transforming my keras model to a tf estimator. With tf.keras.estimator.model_to_estimator( However, I would like to use training Callbacks (mostly for Learning Rate Decay). How can I do this?

**Comment by david-morris** [comment]
I would also like to use this.

I would like to set a callback because there is no keras optimizer which allows you to set a piecewise learning rate, despite this being part of an optimization strategy.
 
I also could not find documentation on how to use a tensorflow optimizer, though I saw [documentation](keras.optimizers.TFOptimizer(optimizer)) suggesting it was possible to wrap a tensorflow optimizer.

This seems to me to be urgent, because of the work of [Wilson et al.](https://arxiv.org/abs/1705.08292) suggesting that adaptive methods may be limited in their applicability.

**Comment by IvanZhangDoIt** [comment]
still no way for this?

**Comment by gabrielilharco** [comment]
+1. What's the recommended approach?

**Comment by zmjjmz** [comment]
I feel like this is more appropriately an issue for the TensorFlow team, so I filed an [issue](https://github.com/tensorflow/tensorflow/issues/21979) there.

**Comment by jerryli1981** [comment]
+1. Any solutions right now?

**Comment by ptiwald** [comment]
If the reason for transforming a keras model to a `tf.estimator` is training on multiple GPUs then you should definitely try the `(tf.)keras.utils.multi_gpu_model` function:
- define `your_keras_model`
- create a new object: `your_keras_model_multi_gpu = multi_gpu_model(your_keras_model, gpus=<number of available GPUs>`
- train `your_keras_model_multi_gpu` just as a standard keras model including all the callback functionality
Just one small warning: when you store weights and you later want to load the weights in a non-multi-gpu model you have to use `your_keras_model.save_weights()`. The weights of `your_keras_model` are updated together while training `your_keras_model_multi_gpu`.

**Comment by rakshithvasudev** [comment]
As @ptiwald mentioned using multi_gpu_model is an option. Unfortunately, this doesn't really fully utilize all the gpu's capacity. In other words, it's slow, effectively defeating the purpose of using multi gpu setup. It's there a workaround to use tf estimator with callbacks in a distributed environment? I don't want to use horovod or other libraries.

**Comment by nareshr8** [comment]
I am trying to use TFRecords to improve performance was thinking about distributed training down the line. I have adaptive learning and early stop callbacks to train the model in keras. Not sure if I can do the same with tf.estimators. Any idea would help.
